src/pages/programDisaggregations/list/ProgramListHooks.ts

```diff
@@ -0,0 +1,93 @@
+import { useDataEngine } from '@dhis2/app-runtime'
+import {
+    useMutation,
+    useQuery,
+    type UseQueryResult,
+} from '@tanstack/react-query'
+import { useCallback, useState } from 'react'
+import { Program } from '../../../types/generated'
+
+export const PROGRAMS_GIST_QUERY_KEY = ['programs-gist'] as const
+interface ProgramsGistResponse {
+    programs: {
+        programs: Program[]
+    }
+}
+
+const fields = ['name', 'displayName', 'id', 'categoryMappings'] as const
+
+export const PROGRAMS_SELECT_QUERY = {
+    resource: 'programs',
+    params: {
+        fields: [...fields],
+    },
+}
+
+export const useProgramsList = (): UseQueryResult<ProgramsGistResponse> => {
+    const engine = useDataEngine()
+
+    return useQuery(PROGRAMS_GIST_QUERY_KEY, async () => {
+        const response = await engine.query({
+            programs: {
+                resource: 'programs/gist',
+                params: {
+                    fields: [...fields],
+                    order: 'name:asc',
+                    pageSize: 200,
+                },
+            },
+        })
+        return response
+    })
+}
```

Comment:
Lets use the same convention as in other places in the app, by using `ResourceQuery` as the querykey, and `useBoundQueryFn` (so you dont have to call engine.query directly)

```ts
const queryFn = useBoundQueryFn()

return useQuery({
   queryKey: [{ 
           resource: 'programs/gist',
           params: {
                 fields: [...fields],
                 order: 'name:asc',
                 pagSize,
            }
      }]
})
```